frontend/src/RegionSearch/ArtistBoroughSearch.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import axios from "axios";
+import { makeStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import { useState, useEffect } from "react";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import { GET_ALL_ARTISTS_API, GET_ALL_BOROUGH_API } from "../constants/constants"
+
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        margin: theme.spacing(0, 0, 3, 0),
+        width: "100%",
+        zIndex: 1,
+        position: "relative",
+        display: "inline-block",
+
+    },
+    element: {
+        margin: theme.spacing(0, 0, 3, 0)
+    }
+}));
+
+
+interface IArtistBoroughSearchProps {
+    defaultArtist?: number;
+    defaultBorough?: string;
+    callbackArtist: (artistId: number | null) => void;
+    callbackBorough: (boroughId: number | null) => void;
+}
+
+
+export default function ArtistBoroughSearch(props: IArtistBoroughSearchProps) {
+    const [artists, setArtists] = useState([]);
+    const [boroughs, setboroughs] = useState([]);
+    const [queryArtist, setQueryArtist] = useState("");
+    const [queryBorough, setQueryBorough] = useState("");
+    const classes = useStyles();
+    const [artistResult, setArtistResult] = useState([]);
+    const [boroughResult, setBoroughResult] = useState([]);
+    const [defaultArtistName, setDefaultArtistName] = useState("");
+    const [defaultBoroughName, setDefaultBoroughName] = useState("");
+
+    useEffect(() => {
+        axios
+            .get(GET_ALL_ARTISTS_API)
```

Comment:
what about parametrizing the `GET_ALL_X_API` URL as a prop? this would let us avoid fetching all artists and boroughs in both instances of the component.